builtin/providers/datadog/resource_datadog_monitor.go

```diff
@@ -0,0 +1,233 @@
+			},
+			"space_aggr": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"operator": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"message": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Alert Settings
+			"warning": &schema.Schema{
+				Type:     schema.TypeMap,
+				Required: true,
+			},
+			"critical": &schema.Schema{
+				Type:     schema.TypeMap,
+				Required: true,
+			},
+
+			// Additional Settings
+			"notify_no_data": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"no_data_timeframe": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func getIDFromResponse(h *http.Response) (string, error) {
+	body, err := ioutil.ReadAll(h.Body)
+	if err != nil {
+		return "", err
+	}
+	h.Body.Close()
+	log.Println(h)
+	log.Println(string(body))
+	v := map[string]interface{}{}
+	err = json.Unmarshal(body, &v)
+	if err != nil {
+		return "", err
+	}
+	if id, ok := v["id"]; ok {
+		return strconv.Itoa(int(id.(float64))), nil
+	}
+	return "", fmt.Errorf("error getting ID from response %s", h.Status)
+}
+
+func marshalMetric(d *schema.ResourceData, typeStr string) ([]byte, error) {
+	name := d.Get("name").(string)
+	message := d.Get("message").(string)
+	timeAggr := d.Get("time_aggr").(string)
+	timeWindow := d.Get("time_window").(string)
+	spaceAggr := d.Get("space_aggr").(string)
+	metric := d.Get("metric").(string)
+	tags := d.Get("metric_tags").(string)
+	operator := d.Get("operator").(string)
+	query := fmt.Sprintf("%s(%s):%s:%s{%s} %s %s", timeAggr, timeWindow, spaceAggr, metric, tags, operator, d.Get(fmt.Sprintf("%s.threshold", typeStr)))
+
+	log.Println(query)
+	m := map[string]interface{}{
+		"type":    "metric alert",
+		"query":   query,
+		"name":    fmt.Sprintf("[%s] %s", typeStr, name),
+		"message": fmt.Sprintf("%s %s", message, d.Get(fmt.Sprintf("%s.notify", typeStr))),
+		"options": map[string]interface{}{
+			"notify_no_data":    d.Get("notify_no_data").(bool),
+			"no_data_timeframe": d.Get("no_data_timeframe").(int),
+		},
+	}
+	return json.Marshal(m)
+}
+
+func authSuffix(meta interface{}) string {
+	m := meta.(map[string]string)
+	return fmt.Sprintf("?api_key=%s&application_key=%s", m["api_key"], m["app_key"])
+}
+
+func resourceMonitorCreate(d *schema.ResourceData, meta interface{}) error {
+	warningBody, _ := marshalMetric(d, "warning")
+	criticalBody, _ := marshalMetric(d, "critical")
+
+	resW, err := http.Post(fmt.Sprintf("%s%s", MONITOR_ENDPOINT, authSuffix(meta)), "application/json", bytes.NewReader(warningBody))
+	if err != nil {
+		return fmt.Errorf("error creating warning: %s", err.Error())
+	}
+
+	resC, err := http.Post(fmt.Sprintf("%s%s", MONITOR_ENDPOINT, authSuffix(meta)), "application/json", bytes.NewReader(criticalBody))
+	if err != nil {
+		return fmt.Errorf("error creating critical: %s", err.Error())
+	}
```

Comment:
We ended up discussing this internally, but because DD doesn't allow multiple thresholds for the same monitor, we ended up going 1:2, where each resource gets two thresholds: warning and critical. Usually we notify on slack only for a warning and on pagerduty for a critical.